tasks/insights-client.yml

```diff
@@ -38,6 +38,62 @@
     insertafter: "#auto_update"
     line: auto_update={{ rhc_insights.autoupdate | d(true) | bool }}
 
+- name: Check ansible host in insights-client config
+  when:
+    - rhc_insights.ansible_host is defined
+    - not rhc_insights.ansible_host is none
+    - rhc_insights.ansible_host != __rhc_state_absent
+    - rhc_insights.ansible_host != omit
+  lineinfile:
+    path: "{{ __rhc_insights_conf }}"
+    regexp: "^ansible_host="
+    state: present
+    line: ansible_host={{ rhc_insights.ansible_host }}
+  check_mode: true
+  register: __insights_ansible_host_exists
+
+- name: Configure ansible host
+  when:
+    - rhc_insights.ansible_host is defined
+    - not rhc_insights.ansible_host is none
+    - rhc_insights.ansible_host != omit
+    - rhc_insights.ansible_host != __rhc_state_absent
+    - __insights_ansible_host_exists.changed
+  block:
+    - name: Update ansible host in insights-client config
+      lineinfile:
+        path: "{{ __rhc_insights_conf }}"
+        regexp: "^ansible_host="
+        insertafter: "#ansible_host="
+        line: ansible_host={{ rhc_insights.ansible_host }}
+    - name: Update ansible host in inventory
+      shell: >-
+        insights-client --ansible-host={{ rhc_insights.ansible_host }} & wait
+      when:
+        - >-
+          "This host is registered" in __rhc_insights_status.stdout
+          or "Registered" in __rhc_insights_status.stdout
+      changed_when: true
+
+- name: Remove ansible host
+  when:
+    - rhc_insights.ansible_host is defined
+    - rhc_insights.ansible_host == __rhc_state_absent
+  block:
+    - name: Remove ansible host from insights-client config
+      lineinfile:
+        path: "{{ __rhc_insights_conf }}"
+        regexp: "^ansible_host="
+        state: absent
+    - name: Remove ansible host in inventory
+      shell: insights-client --ansible-host= & wait
```

Comment:
Running `insights-client --ansible-host=` really resets the Ansible Host value in the Inventory. Removing it from the config file, or setting it there to `ansible_host=`, however, doesn’t do so: after an upload, the value remains unchanged.

That means, the two changes do different things:

* The change to the configuration file causes Ansible Host to remain unchanged in the Inventory, regardless of what the stored value is.
* The CLI call causes the Ansible Host value to reset in the inventory.

Reply:
> Is it ok that it is not possible to reset the Ansible Host value with the config file?

I'm not sure what you mean - do you mean "without using the `rhc` role"?

Reply:
Yes yes, using plain Insights Client CLI. I am aware that this is not the place to ask, it just came up on my mind.